src/classes/RD2_EntryFormController.cls

```diff
@@ -354,24 +304,83 @@ public with sharing class RD2_EntryFormController {
             set;
         }
 
-        /***
-        * @description Builds an Elevate Commitment request body
-        * @param recordId Recurring Donation Id
+        /**
+        * @description Handles validation and Elevate recurring commitment creation
+        * @param jsonRecord Recurring Donation fields as entered by the user before the record is saved
         * @param paymentMethodToken Payment Method Token
-        * @return PS_CommitmentRequest.RequestBody
+        * @return UTIL_Http.Response Payments API response
         */
-        public PS_CommitmentRequest.RequestBody getRequestBody(Id recordId, String paymentMethodToken) {
-            PS_CommitmentRequest request = new PS_CommitmentRequest();
+        public UTIL_Http.Response handleCommitment(String jsonRecord, String paymentMethodToken) {
```

Comment:
The code added in this WI is very similar to Edit Commitment POC #6142, but it did diverge a bit. Thus, when edit commitment feature is being implemented, please do not copy code over `rd2EntryForm`, and `RD2_EntryFormController` but feel free to add pieces here and there understanding the copied and changed code.